pkg/antctl/raw/set/command.go

```diff
@@ -0,0 +1,181 @@
+			return setBoolOrFail(&c.ClickHouse.Debug, value)
+		},
+		"clickhouse.compress": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			if c.ClickHouse.Compress == nil {
+				c.ClickHouse.Compress = new(bool)
+			}
+			return setBoolOrFail(c.ClickHouse.Compress, value)
+		},
+		"clickhouse.commitInterval": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setTimeDurationOrFail(&c.ClickHouse.CommitInterval, value)
+		},
+		"clickhouse.ttl": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setIntOrFail(&c.ClickHouse.TTL, value)
+		},
+		"flowcollector.enable": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setBoolOrFail(&c.FlowCollector.Enable, value)
+		},
+		"flowcollector.externalflowcollectoraddr": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setStringOrFail(&c.FlowCollector.Address, value)
+		},
+	}
+	Command.Example = example
+	Command.RunE = updateRunE
+}
+
+func updateRunE(cmd *cobra.Command, args []string) error {
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	k8sClient, _, err := raw.SetupClients(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to create clientset: %w", err)
+	}
+	configMap, err := flowaggregatorconfig.GetFAConfigMap(k8sClient)
+	if err != nil {
+		return err
+	}
+	var flowAggregatorConf flowaggregatorconfig.FlowAggregatorConfig
+	if err := yaml.Unmarshal([]byte(configMap.Data["flow-aggregator.conf"]), &flowAggregatorConf); err != nil {
+		klog.Errorf("Failed to unmarshal FlowAggregator config from ConfigMap: %v", err)
+		return err
+	}
+	for _, query := range args {
+		pair := strings.Split(query, "=")
+		if len(pair) != 2 {
+			return fmt.Errorf("query should contains exactly one '='")
+		}
+		if mutatorFn, ok := mutators[pair[0]]; ok {
+			err := mutatorFn(&flowAggregatorConf, pair[1])
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("wrong arg, please check antctl set -h")
+		}
+	}
+	// marshal back the changed parameters to configmap
+	b, err := yaml.Marshal(&flowAggregatorConf)
+	if err != nil {
+		klog.Warningf("Failed to marshal FlowAggregator config")
+		return err
+	}
+	configMap.Data["flow-aggregator.conf"] = string(b)
+	if _, err := k8sClient.CoreV1().ConfigMaps("flow-aggregator").Update(context.TODO(), configMap, metav1.UpdateOptions{}); err != nil {
+		klog.Warningf("Failed to update ConfigMap %s: %v", configMap.Name, err)
+		return err
+	}
+	return nil
+}
+
+func setBoolOrFail(b *bool, value string) error {
+	if value != "true" && value != "false" {
+		return fmt.Errorf("value should be true or false")
+	}
+	*b = value == "true"
+	return nil
+}
+
+func setStringOrFail(b *string, value string) error {
+	*b = value
+	return nil
+}
+
+func setTimeDurationOrFail(b *string, value string) error {
```

Comment:
I don't think it makes sense to have this as a generic function, since the implementation is specific to this config parameter (the commit interval). Either rename this function to `setCommitIntervalOrFail` or don't write a helper function at all and move this logic to line 86.

Reply:
I've renamed it.